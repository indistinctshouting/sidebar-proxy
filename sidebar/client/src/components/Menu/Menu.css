.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(36, 35, 35, 0.6);
}

.displayblock {
  display: block;
}

.displaynone {
  display: none;
}

.modalmain {
  position: fixed;
  background: white;
  min-width: 500px;
  min-height: 200px;
  max-height: calc(95vh - 200px);
  width: 50%;
  top:50%;
  left:50%;
  overflow-y: auto;
  transform: translate(-50%,-50%);
  padding: 30px;
  border-radius: 20px
}

.bluetext {
  color: #0073BB;
}

.menuimage {
  border-radius: 10%;
}