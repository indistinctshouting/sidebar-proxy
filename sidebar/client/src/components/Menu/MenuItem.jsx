import React from 'react';
import { Media } from 'react-bootstrap';
import style from './Menu.css';

let emptyImage = "https://s3-media2.fl.yelpcdn.com/assets/2/www/img/dca54f97fb84/default_avatars/menu_medium_square.png";

const MenuItem = (props) => {
  return (
    <div>
      <hr/>
      <Media>
        <Media.Left align="middle">
          <img className={style.menuimage} width={70} height={70} src={emptyImage} alt="thumbnail"/>
        </Media.Left>
        <Media.Body>
          <Media.Heading componentClass="h5" bsClass="pull-left">
            <strong className={style.bluetext}>
              {props.details.item_name}
            </strong>
        </Media.Heading>
          <Media.Heading componentClass="h5" bsClass="pull-right">
            <strong>
              {props.details.item_price}
            </strong>
        </Media.Heading>
          <br></br>
          <br></br>
          <p>{props.details.item_description}</p>
        </Media.Body>
      </Media>
    </div>
  )
}
export default MenuItem
