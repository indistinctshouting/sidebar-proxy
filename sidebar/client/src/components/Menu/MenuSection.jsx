import React from 'react';
import MenuItem from './MenuItem.jsx'

const MenuSection = (props) => {
  return (
    <div>
      <h4>
        {props.section}
      </h4>
      {props.items.map(item => 
        <MenuItem 
          key={item.item_name} 
          details={item}
        />
      )}
      <br></br>
    </div>
  )
}

export default MenuSection;