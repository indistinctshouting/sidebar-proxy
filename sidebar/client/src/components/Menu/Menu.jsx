import React from 'react';
import MenuSection from './MenuSection.jsx';
import style from './Menu.css';

const Menu = (props) => {
  const showOrHideClass = (props.show) ? `${style.modal} ${style.displayblock}` : `${style.displaynone}`;
  return (
    <div className={showOrHideClass}>
      <section className={style.modalmain}>
        <h4>
          Menu For {props.name}
          <button type="button" className="close" aria-label="Close" onClick={props.hideMenu}>
            <span>&#10005;</span>
          </button>
          <hr />
        </h4>
        {Object.keys(props.menuInfo).map(section =>
          <MenuSection
            key={section}
            section={section}
            items={props.menuInfo[section]}
          />
        )}
        <button onClick={props.hideMenu}>Close</button>
      </section>
    </div>
  )
}

export default Menu;