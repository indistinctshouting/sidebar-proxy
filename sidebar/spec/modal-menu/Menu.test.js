import React from 'react';
import { shallow, mount } from 'enzyme';
import Menu from '../../client/src/components/Menu';

describe('Menu', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Menu 
        show={false} 
        menuInfo={{}} 
        name={'restaurant'} 
        hideMenu={() => { return; }}
        />
    )
  })

  it('should render menu component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an initial div container with class display-none', () => {
    let divContainer = wrapper.find('.display-none');
    expect(divContainer.hasClass('display-none')).toEqual(true);
  });

  it('should have the restaurant name in the header', () => {
    let header = wrapper.find('h4');
    expect(header.text()).toMatch(/Menu For restaurant/);
  });

  it('should have a close button', () => {
    let closeButton = wrapper.find('.close');
    expect(closeButton).toBeDefined();
  });

  it('should not have MenuSection components if the menu object is empty', () => {
    let menuSections = wrapper.find('MenuSection');
    expect(menuSections.exists()).toEqual(false);
  });

  it('should have the correct number of MenuSection components', () => {
    let wrapper = shallow(
      <Menu 
        show={false} 
        menuInfo={{1: {}, 2: {}}} 
        name={'restaurant'} 
        hideMenu={() => { return; }}
      />
    )

    let menuSections = wrapper.find('MenuSection');
    expect(menuSections.length).toEqual(2);
  })
  
});
