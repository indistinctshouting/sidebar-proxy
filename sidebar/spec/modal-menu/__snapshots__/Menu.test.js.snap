// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render menu component correctly 1`] = `
<div
  className="display-none"
>
  <section
    className="modal-main"
  >
    <h4>
      Menu For 
      restaurant
      <button
        aria-label="Close"
        className="close"
        onClick={[Function]}
        type="button"
      >
        <span>
          âœ•
        </span>
      </button>
      <hr />
    </h4>
    <button
      onClick={[Function]}
    >
      Close
    </button>
  </section>
</div>
`;
